contracts/RadspecRegistry.sol

```diff
@@ -359,17 +360,31 @@ contract RadspecRegistry is IArbitrable, AragonApp {
     /**
     * @dev Give a ruling for a certain dispute, the account calling it must have rights to rule on the contract
     * @param _disputeId Identification number of the dispute to be ruled
-    * @param _ruling Ruling given by the arbitrator, where 0 is reserved for "refused to make a decision"
+    * @param _ruling Ruling given by the arbitrator, where 0, 1 and 2 are reasons to void the dispute, 3 represents
+    *        a failed dispute and 4 represents a successful dispute.
     */
-    function rule(uint256 _disputeId, uint256 _ruling) external {
-        // TODO Error messages
-        require(msg.sender == arbitrator);
-        require(_disputeExists(_disputeId));
+    function rule(uint256 _disputeId, uint256 _ruling) external nonReentrant {
+        require(msg.sender == address(arbitrator), ERROR_NOT_ARBITRATOR);
+        require(_disputeExists(_disputeId), ERROR_DISPUTE_DOESNT_EXIST);
 
-        // TODO Transfer stake to plaintiff
-        // TODO Remove entry
-        // TODO Remove dispute
+        DisputeInfo storage disputeInfo = disputes[_disputeId];
+        Entry storage entry = entries[disputeInfo.scope][disputeInfo.sig];
+
+        if (_ruling == 4) {
```

Comment:
Maybe we should have a constant somewhere to make it clearer what `_ruling == 4` actually means.